### PR TITLE
Introduce keywords for actions to filter on (for #4679)

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

**Comment by bpasero** [comment]
@alexandrudima maybe you could give me some feedback on https://github.com/Microsoft/vscode/pull/5432/commits/eebd22d93e82aa03e635e74ae8134d8159c8905b

The motivation is to be able to provide an array of keywords when registering an editor action so that we have means to find the action in the command palette in addition to the label of the action. We can then pass in english (non translated) words to fix https://github.com/Microsoft/vscode/issues/4679

**Comment by outcoldman** [comment]
@bpasero just small suggestion, you can use `#4679` instead of `4679` to have a reference/link on pull request and commit history.

**Comment by bpasero** [comment]
👍

File: src/vs/workbench/common/actionRegistry.ts [hunk 1]
```diff
@@ -52,24 +54,40 @@ export interface IWorkbenchActionRegistry {
...
+	public registerWorkbenchAction(descriptor: SyncActionDescriptor, keywords?: string[]): void;
+	public registerWorkbenchAction(descriptor: SyncActionDescriptor, category?: string, keywords?: string[]): void;
+	public registerWorkbenchAction(descriptor: SyncActionDescriptor, categoryOrKeywords?: string|string[], keywords?: string[]): void {
 		if (!this.workbenchActions[descriptor.id]) {
 			this.workbenchActions[descriptor.id] = descriptor;
 			registerWorkbenchCommandFromAction(descriptor);
 
-			if (category) {
-				this.mapActionIdToCategory[descriptor.id] = category;
+			if (typeof categoryOrKeywords === 'string') {
```
Comment (outcoldman) [inline src/vs/workbench/common/actionRegistry.ts]: `Types.isString`?

File: src/vs/workbench/common/actionRegistry.ts [hunk 2]
```diff
@@ -52,24 +54,40 @@ export interface IWorkbenchActionRegistry {
...
 
 class WorkbenchActionRegistry implements IWorkbenchActionRegistry {
 	private workbenchActions: collections.IStringDictionary<SyncActionDescriptor>;
 	private mapActionIdToCategory: { [id: string]: string; };
+	private mapActionIdToKeywords: { [id: string]: string[]; };
 
 	constructor() {
 		this.workbenchActions = Object.create(null);
 		this.mapActionIdToCategory = Object.create(null);
+		this.mapActionIdToKeywords = Object.create(null);
```
Comment (outcoldman) [inline src/vs/workbench/common/actionRegistry.ts]: will it be better to actually have a map to everything? Have map where values will be some structure, so you will not need to maintain two of them?

**Comment by outcoldman** [comment]
no unit tests?

**Pull request merged**